vyper/builtin_functions/functions.py

```diff
@@ -1302,6 +1302,22 @@ def build_IR(self, expr, args, kwargs, contact):
         )
 
 
+class RawRevert(BuiltinFunction):
+
+    _id = "raw_revert"
+    _inputs = [("data", BytesAbstractType())]
+
+    def fetch_call_return(self, node):
+        return None
+
+    @process_inputs
+    def build_IR(self, expr, args, kwargs, context):
+        error_data_buf = ensure_in_memory(args[0], context)
+        data = bytes_data_ptr(error_data_buf)
```

Comment:
i think that the error_data_buf needs to be cached.
```
with ensure_in_memory(args[0], context).cache_when_complex("err_buf") as error_data_buf:
    ...
```

Reply:
Added caching